documentation/modules/exploit/linux/local/tomcat_rhel_based_temp_priv_esc.md

```diff
@@ -0,0 +1,148 @@
+## Vulnerable Application
+
+This module exploits a vulnerability in RedHat based systems where
+improper file permissions are applied to `/usr/lib/tmpfiles.d/tomcat.conf`
+for Apache Tomcat versions before 7.0.54-8.  This may also work against
+
+The configuration files in `tmpfiles.d` are used by `systemd-tmpfiles` to manage
+temporary files including their creation.
+
+With this weak permission, we're able to inject commands into `systemd-tmpfiles`
+service to write a cron job to execute our payload.
+
+`systemd-tmpfiles` is executed by default on boot on RedHat-based systems
+through `systemd-tmpfiles-setup.service`. Depending on the system in use,
+the execution of `systemd-tmpfiles` could also be triggered by other
+services, cronjobs, startup scripts etc.
```

Comment:
Might be misunderstanding but thought I'd ask, I'm just wondering how module cleanup would work in the scenario of the computer restarting and resulting in the original session being killed?

Reply:
it would need a manual cleanup. the payload by default goes to `/tmp` so that would be gone, so just remove a cron entry on re-exploitation.